### PR TITLE
tests: Clean cmath and climits usage

### PR DESCRIPTION
**Pull request by krOoze**
- remove usage of `limits.h` that should not have existed
- merge the two separate handling of `nextafter` direction
- remove workaround for `cmath` impls with missing `std::`

deprecates PR #2443

### CODE REVIEW

**Comment by jzulauf-lunarg** [comment]
My vague memory was that I had used the limits.h versions based on std:: portability issues with Android.  Building on our internal CI.  @cnorthrop  -- do you have bandwidth to do Android sanity testing on this?

**Comment by krOoze** [comment]
@jzulauf-lunarg `limits.h` is completely unnecessery as we do **not** use plain `int` and `unsigned`.

But yeah, standard library standardness was a known problem on Android, and it affects `<math.h>` vs `<cmath>`
Thought it was time to start removing workarounds (as public CI seems fine with this now)...

**Comment by jzulauf-lunarg** [comment]
Clean on our CI as well, I'd like to wait on Cody @cnorthrop  for the approval on this, just in case there are compile and target combinations CI isn't covering.

BTW -- change is a nice simplification.

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -6419,9 +6403,9 @@ TEST_F(VkLayerTest, CreatePipelineLayoutExcessDescriptorsOverall) {
     uint32_t sum_samplers = m_device->phy().properties().limits.maxDescriptorSetSamplers;
     uint32_t sum_input_attachments = m_device->phy().properties().limits.maxDescriptorSetInputAttachments;
 
-    // Devices that report UINT_MAX for any of these limits can't run this test
-    if (UINT_MAX == std::max({sum_dyn_uniform_buffers, sum_uniform_buffers, sum_dyn_storage_buffers, sum_storage_buffers,
-                              sum_sampled_images, sum_storage_images, sum_samplers, sum_input_attachments})) {
+    // Devices that report UINT32_MAX for any of these limits can't run this test
+    if (UINT32_MAX == std::max({sum_dyn_uniform_buffers, sum_uniform_buffers, sum_dyn_storage_buffers, sum_storage_buffers,
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: good catch on the UINT vs. UINT32

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -10277,7 +10261,7 @@ TEST_F(VkLayerTest, PSOLineWidthInvalid) {
     gp_ci.renderPass = renderPass();
     gp_ci.subpass = 0;
 
-    const std::vector<float> test_cases = {-1.0f, 0.0f, nextafterf(1.0f, 0.0f), nextafterf(1.0f, 2.0f), NAN};
+    const std::vector<float> test_cases = {-1.0f, 0.0f, NearestSmaller(1.0f), NearestGreater(1.0f), NAN};
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: That's clearer.

File: tests/layer_validation_tests.cpp [hunk 3]
```diff
@@ -21919,14 +21903,14 @@ TEST_F(VkLayerTest, ScissorBoundsChecking) {
 
     {
         m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, VALIDATION_ERROR_1d8004a8);
-        VkRect2D scissor = {{100, 100}, {INT_MAX, 16}};
+        VkRect2D scissor = {{100, 100}, {INT32_MAX, 16}};
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: :+1:

**Comment by cnorthrop** [comment]
Looks good from our end, although we're still at partial test coverage due to our office move.  We can revisit if it becomes a problem when we get remaining systems back online.

**Pull request merged**